### ISSUE TITLE
Restricting a phenotype type to a specific genotype type

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#1471]
(Required for #1462)

Currently, there is no restriction on which phenotype can be used to annotate a genotype, meaning that it is possible (for example) to annotate a host-only genotype with a pathogen phenotype.

We need to add a constraint that limits which phenotype annotations can apply to different types of genotype.

@kimrutherford suggested adding an optional property to the annotation types in the configuration file – the property could be called `feature_subtype`, and will have three possible values, corresponding to the three phenotype types in PHIPO:

- `pathogen`
- `host`
- `pathogen_host_interaction`

Where `pathogen_host_interaction` corresponds to a _disease formation phenotype_.

**Comment by kimrutherford** [comment]
I think this depends on #1442.

**Comment by jseager7** [comment]
Is that because we currently don't have a way to test if a genotype is a `pathogen_host_interaction`, i.e. a meta-genotype?

**Comment by kimrutherford** [comment]
> Is that because we currently don't have a way to test if a genotype is a pathogen_host_interaction,
> i.e. a meta-genotype?

It's more that we don't have a was of storing a meta-genotype.  Currently if you configure some hosts and then make a genotype with host and pathogen alleles it gets stored as a normal genotype.  We need more information in the database to tell the difference between the different sorts of genotype.  The main issue we the current schema is that there's no way to store a meta-genotype where there are no host alleles because it looks just like a pathogen genotype.

**Comment by jseager7** [comment]
I've tested this on my virtual machine and there don't seem to be any problems, besides the fact I'm unable to annotate a host genotype with a phenotype because the only available term is 'Host phenotype', and that shouldn't be used for direct annotations &ndash; but that's the intended behaviour.

I've tested creating a pathogen phenotype and a disease formation phenotype, and that works fine. I've also confirmed that the evidence codes for those phenotypes match what we have in the evidence code document.

One slightly inconsistent thing is that the root terms 'Host phenotype' and 'Pathogen phenotype' appear in their respective term selections, but 'Disease formation phenotype' does not appear in the term selection for meta-genotypes. Not sure if that's a problem or not.

**Comment by kimrutherford** [comment]
Thanks for testing.

> One slightly inconsistent thing is that the root terms 'Host phenotype' and 'Pathogen phenotype' appear
> in their respective term selections, but 'Disease formation phenotype' does not appear in the term
> selection for meta-genotypes.

The disease formation phenotype term is marked as "do not annotate" in the OBO file:

    subset: qc_do_not_annotate
    subset: qc_do_not_manually_annotate

The other two aren't.

I'll make a separate issue about that.

**Comment by kimrutherford** [comment]
> I'll make a separate issue about that.

PHI-base/phipo#19
